charts/trino/templates/configmap-coordinator.yaml

```diff
@@ -46,7 +46,9 @@ data:
     http-server.http.port={{ .Values.service.port }}
     query.max-memory={{ .Values.server.config.query.maxMemory }}
     query.max-memory-per-node={{ .Values.server.config.query.maxMemoryPerNode }}
+{{- if (and (ne (toString .Values.image.tag) "latest") (lt (int .Values.image.tag) 369)) }}
```

Comment:
Just to point out one thing that I don't like. With this we are essentially forcing user of this chart to use images tagged with numbers (with 369 being significant). Even main trinodb/trino dockerhub image has un-bundled tags like `369-arm64` also anyone that adds/modifies this image and retags it with whatever different name will not be able to use this chart.
So in my opinion it is better to have 2 versions of chart, old that works pre 369 and new that works after then make chart work only with forced tag template.  
WDYT @hashhar @hovaesco ?

Reply:
That is a good point that I had not considered.

Would a simpler solution be to render this config only if the value is provided?
i.e. change default value to `null` or empty value. Render only non-empty values in ConfigMap. Add a description for the config that explains to leave it unset when using Trino >= 369?

Reply:
Having two separate versions of the chart would be harder to maintain, manage etc.

@hashhar approach sounds good. I would also comment `maxTotalMemoryPerNode` by default with the suggested comment from above.

Reply:
Sounds good @hashhar 
With that change user would only need to remove `maxTotalMemoryPerNode` form they deployment files to be able to use versions >369.

Reply:
The problem with leaving it in for the sake of backwards compatibility is that users are going to update to the latest chart and potentially have a broken deployment.  In my experience with other projects, this usually results in a bunch of issues/support requests being needlessly generated.

In my eyes, it makes the most sense to remove it entirely and bump the chart version to `0.4.0` as a breaking change.  Users who need to stay on Trino <369 should continue to use the `0.3` chart.  This should result in very little/no ongoing maintenance.

Yes, this will likely also result in a few broken deployments when users upgrade their helm chart to `>=0.4` and pin their Trino tag to an old version.  Seems favorable as opposed to adding the additional complexity to the chart to and future maintenance.

Reply:
I'm not very familiar with how people interact with Helm. If chart updates aren't applied automatically and _**if Helm can catch that there are some unused values provided in deployment (i.e. keys that aren't part of values.yml)**_ then what @erikcw is suggesting is also fine.

I don't think there are going to be any huge changes in chart so people can continue using older chart until they reach Trino 369?

@electrum any opinion?